src/wallet/home_cards/manager.rs

```diff
@@ -99,6 +99,19 @@ impl HomeCardsManager {
         self.save_cards(updated_cards).await
     }
 
+    /// Marks the wallet restoration.
+    /// Ensures only the expected `HomeCards` remain in `HomeCardsStorage` - currently none.
+    /// Notifies `HomeCardsObserver`.
+    #[uniffi::method]
+    pub async fn wallet_restored(&self) -> Result<()> {
+        let updated_cards = self
+            .update_cards(|write_guard| {
+                **write_guard = HomeCards::new();
+            })
+            .await?;
+        self.save_cards(updated_cards).await
```

Comment:
Should we put this in a private method that `wallet_restored` & `wallet_reset` call?